src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py

```diff
@@ -155,7 +161,7 @@ def get_completions(self, document, complete_event):
             yield comp
 
         if self.complete_command and self.cmdtab and self.leftover_args and self.leftover_args[-1].startswith('-'):
-            for comp in sort_completions(self.gen_dynamic_completions(text)):
+            for comp in sort_completions(self.gen_dynamic_completions(text_split)):
```

Comment:
The reason why I changed this from `text` to `text_split` is that `text_split` might be altered in `EVENT_INTERACTIVE_PRE_SUB_TREE_CREATE` event hook, and since `text` is an immutable string, I cannot alter `text` from the event hook. I re-created `text` by joining `text_split` and appending `self.unfinished_word` in `gen_dynamic_completions`.

Reply:
Did you consider passing `text` along inside a dictionary? That way you can manipulate whatever data you want inside and then access it again after the event handler is called.

Reply:
Updated - https://github.com/chewong/azure-cli-alias-extension/blob/tab-completion/azext_alias/hooks.py#L82-L96